### PR TITLE
COV-42: update address schema to match domain model

### PR DESCRIPTION
**Pull request by jason-gerard**
JIRA: https://soen-390-covid-tracker.atlassian.net/browse/COV-42

### CODE REVIEW

File: server/db/schemas/addresses.sql [hunk 1]
```diff
@@ -1,11 +1,18 @@
...
-    FOREIGN KEY (user_id) REFERENCES users (user_id)
+    PRIMARY KEY (address_id),
+    UNIQUE (
+        street_address,
+        street_address_line_two,
+        city,
+        province,
+        postal_code,
+        country
+   )
```
Comment (jason-gerard) [inline server/db/schemas/addresses.sql:17]: This was added so we can uniquely identify matching addresses without knowing the primary key and so there cannot be duplicate addresses

File: server/db/schemas/addresses.sql [hunk 2]
```diff
@@ -1,11 +1,18 @@
 CREATE TABLE addresses (
-    address_id serial PRIMARY KEY,
-    user_id INT NOT NULL,
+    address_id SERIAL,
     street_address VARCHAR(255) NOT NULL,
-    street_address_line_two VARCHAR(255),
+    street_address_line_two VARCHAR(255) NOT NULL,
```
Comment (domsec) [inline server/db/schemas/addresses.sql:4]: `street_address_line_two` is optional so can be null.

File: server/db/schemas/addresses.sql (hunk 1 above)
Comment (domsec) [reply server/db/schemas/addresses.sql:17]: good idea

**Review by domsec — APPROVED** [review]

File: server/db/schemas/addresses.sql (hunk 2 above)
Comment (jason-gerard) [reply server/db/schemas/addresses.sql:4]: If the attribute is null you can't use it in a `UNIQUE` constraint and so 2 people in an apartment building for example would have overlapping addresses.

Comment (domsec) [reply server/db/schemas/addresses.sql:4]: I figured as much but wasn't sure. Thanks for the clarification.

**Pull request merged**